### DISCUSSION TITLE
TextLayer text size in 3D

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#6944] [Q&A]
Hi, I have a question regarding to font size in TextLayer. I have some points (text labels) and 3D model in some altitude (100m +)

Now I have `viewState.position = [ 0, 0, 0]` (default) and all labels are too huge (pixels)
![image](https://user-images.githubusercontent.com/23722745/169809633-5320ebbc-f6c3-419a-9c05-c7be7a29aafc.png)

Now I have `viewState.position = [ 0, 0, some altitude from 3D data]` (when I zoom in labels are too small, almost invisible)
![image](https://user-images.githubusercontent.com/23722745/169810164-4ef5e408-7eb7-4f6c-b5c7-a979166fa914.png)

I think it's all related to `viewState.position` with Z. When i set `sizeMinPixels`, `sizeMaxPixels` it does not work it seems, labels become almost invisible if you zoom closer to the model.

There is much more bugs/my bad implementations with 3D data with some elevation. I want to rotate around model, or terrain if used, not around Z=0.

I played with cairo data (https://loaders.gl/examples/3d-tiles) and set altitude automatically from tileset (just an example)

```
onTilesetLoad: (tileset) => {
viewState.position = [ 0, 0, tileset.cartographicCenter[2]]
}
```

And data disappear in some distance

![image](https://user-images.githubusercontent.com/23722745/169812737-52c8653c-0167-47c4-b75e-e7226a49d7d2.png)

sometimes playing with `farZMultiplier ` helps, but  it does not solve disappearing tiles at all.

I started with labels in TextLayer, but it seems to be deeper issue. Do you have some recommendations how to work with tiles and viewstate position with non 0 anchor point?

Thanks!

**Comment by hkfb** [comment]
Related discussion: https://github.com/visgl/deck.gl/discussions/6808

And related issue: https://github.com/visgl/deck.gl/issues/6925

**Reply by radoslavirha** [reply]
Yeah but all the issues are talking about OrbitView. I need MapView. I always combine 3D data with some basemap and other data.
